java/client/src/org/openqa/selenium/remote/mobile/RemoteNetworkConnection.java

```diff
@@ -42,7 +42,7 @@ public ConnectionType getNetworkConnection() {
   @Override
   public ConnectionType setNetworkConnection(
       ConnectionType type) {
-    Map<String, ConnectionType> mode = ImmutableMap.of("type", type);
+    Map<String, Integer> mode = ImmutableMap.of("type", type.getBitMask());
```

Comment:
looking at it again, we can just add the `toJson()` method to ConnectionType and then the JsonToBeanConverter will do it's thing and serialize it.

    public String toJson(){ return toString(); }